main.cpp

```diff
@@ -19,13 +19,34 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+  CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) {
+  };
+		
+  virtual void run(const MatchFinder::MatchResult& Result) {
+	  const auto *CastExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+    auto &SM = *Result.SourceManager;
+    auto DestTypeString = Lexer::getSourceText(CharSourceRange::getTokenRange(
+                                               CastExpr->getLParenLoc().getLocWithOffset(1),
+                                               CastExpr->getRParenLoc().getLocWithOffset(-1)),
+                                               SM,
+  		  																			 Result.Context->getLangOpts()
+    );
+    auto s = ("static_cast<" + DestTypeString + ">(").str();
+    auto Range = CharSourceRange::getCharRange(CastExpr -> getLParenLoc(),
+ 			                                         CastExpr -> getSubExprAsWritten()->getBeginLoc()
+    );
+    rewriter_.ReplaceText(Range, s);
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
-    }
+ 	  const auto *SubExpr = CastExpr->getSubExprAsWritten()->IgnoreImpCasts();
```

Comment:
Please describe what happens if we change the code as follows:  
```suggestion
 	  const auto *SubExpr = CastExpr->getSubExprAsWritten();
```
What is the purpose of using the IgnoreImpCasts() method?

Reply:
С помощью метода `IgnoreImpCasts()` игнорируются все неявные приведения типов, которые могут окружать выражение
Но если убрать этот метод, то в данном случае ничего не изменится, т.к. в примере нет неявного приведения

Reply:
If you try to print the AST of the `test.cpp` example, you will see that you have several `ImplicitCastExpr` nodes:
![image](https://user-images.githubusercontent.com/76072618/119860568-20cb0700-bf1f-11eb-8e3d-2a533baa265f.png)

Depending on which approaches you use to get the subexpression node, you will get different result nodes for the `test.cpp` example:

Code:
```cpp
// getSubExpr() retrieves the first cast subexpression
llvm::outs() << "getSubExpr(): " << castNode->getSubExpr()->getStmtClassName();

// getSubExprAsWritten() retrieves the cast subexpression as it was written in the source
// code, looking through any implicit casts or other intermediate nodes
// introduced by semantic analysis.
llvm::outs() << "getSubExprAsWritten(): " << castNode->getSubExprAsWritten()->getStmtClassName();

// getSubExpr() retrieves the first cast subexpression.
// ignoreImpCasts() skips past any implicit casts which might surround this expression until
// reaching a fixed point. It skips the following AST nodes: ImplicitCastExpr, FullExpr.
llvm::outs() << "getSubExpr()->ignoreImpCasts(): " << castNode->getSubExpr()->IgnoreImpCasts()->getStmtClassName();
```
Output:
```bash
 getSubExpr(): ImplicitCastExpr 
 getSubExprAsWritten(): DeclRefExpr  
 getSubExpr()->ignoreImpCasts(): DeclRefExpr
```